Fixed #33046 -- Documented fully cached query approach of count. (#14799)
django/django: docs/ref/models/querysets.txt

probably more efficient to use ``len(queryset)`` which won't cause an extra
database query like ``count()`` would.

> .. note::
>     If the queryset has already been fully retrieved, count will use that length
>     rather than perform an additional query.
>     
>     The following code hits the database once::
> 
>     >>> pizzas = Pizza.objects.prefetch_related('toppings')
>     >>> {pizza.name:pizza.toppings.count() for pizza in pizzas}
> 

Reviewer: There is no need for a separate note and an example, IMO. Maybe:
```suggestion
If the queryset has already been fully retrieved, ``count()`` will use that
length rather than perform an extra database query.
```
